### ISSUE TITLE
"Max" button doesn't work if a token with spending limit was previously selected

### ISSUE THREAD

**Issue by francovenica** [safe-global/safe-wallet-web#3326]
## Bug description
TL;DR: Check the gif

In send funds, the "Max" sets the value to the max amount you have of a specific token or the max allowed by the spending limit created for that user.
If you have a token that has a spending limit, select using it, then switch to another token that doesn't have a spending limit set for the user and use the "Max" button, the amount would remain at 0
I assume that, even the spending limit option is not being displayed, is still "selected" for that new token

## Environment
 - Browser: Chrome
 - Wallet: MetaMask
 - Chain: Ethereum mainnet
 - Env: Current prod 1.30

## Steps to reproduce
 1. Go to safe where you have 2 tokens
 2. Create a spending limit for one of them, for the user you are corrently connected to
 3. Go to send funds
 4. Select the token you created the spending limit for
 5. Select "use spending limit"
 6. Switch to the other token with the token selector
 7. Click max

## Expected result
It should autocomplete the amount input to the max of that token

## Obtained result
Sets a 0

## Screenshots
![max button](https://github.com/safe-global/safe-wallet-web/assets/4503659/20828451-5229-49b0-b3a8-173330ba861c)
